bec_server/bec_server/device_server/rpc_mixin.py

```diff
@@ -156,6 +156,25 @@ def _handle_misc_rpc(self, instr: messages.DeviceInstructionMessage) -> Any:
                 instr_params.get("func").split(".put")[0],
             )
             self._update_cache(obj, instr)
+        elif instr_params.get("func") == "set" or instr_params.get("func").endswith(".set"):
+            if instr_params.get("func") == "set":
+                obj = self.device_manager.devices[device_root].obj
+            else:
```

Comment:
This is a little hard to read, and it might be nicer in its own function rather than "misc" RPC? this whole thing could use a little refactoring with a helper func to get the correct device obj since that is repeated quite a few times here for the different operations